File: User/lib/network/response_models/intro_model.dart
class IntroModel {
  String? image;
  bool? status;
  String? clientId;
  String? id;
  String? tittle;
  String? description;

  IntroModel(
      {this.image,
        this.status,
        this.clientId,
        this.id,
        this.tittle,
        this.description});

  IntroModel.fromJson(Map<String, dynamic> json) {
    image = json['image'];
    status = json['status'];
    clientId = json['clientId'];
    id = json['id'];
    tittle = json['title'];
    description = json['description'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['image'] = image;
    data['status'] = status;
    data['clientId'] = clientId;
    data['id'] = id;
    data['title'] = tittle;
    data['description'] = description;
    return data;
  }
}

File: Backend/src/models/client/intro.model.js
const mongoose = require('mongoose');
const { toJSON, paginate } = require('../plugins');

const introSchema = mongoose.Schema(
  {
    title: {
      type: String,
      required: false,
    },
    description: {
      type: String,
      required: false,
    },
    type: {
      type: String,
      required: false,
    },
    image: {
      type: String,
      required: true,
    },
    status: {
      type: Boolean,
      required: false,
    },
    clientId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Client',
    },
  },
  {
    timestamps: true,
  },
);

// add plugin that converts mongoose to json
introSchema.plugin(toJSON);
introSchema.plugin(paginate);

/**
 * @typedef IntroImage
 */
const IntroImage = mongoose.model('IntroImage', introSchema);

module.exports = IntroImage;
